### PR TITLE
refactor(search): re-works tabs for clarity

### PR DESCRIPTION
**Pull request by dzucconi**
Previously there was a function `renderTab` declared in the body of the parent component. In unwinding this I took the liberty to clean up the logic substantially to make this more legible. I also fixed some spacing issues that were causing a non-system vertical rhythm on the page.

### CODE REVIEW

File: src/Apps/Search/Components/NavigationTabs.tsx [hunk 1]
```diff
@@ -1,166 +1,217 @@
...
-import { useRouter } from "System/Router/useRouter"
 import { AppContainer } from "Apps/Components/AppContainer"
 import { useTracking } from "react-tracking"
+import { FC, useMemo } from "react"
+import { compact } from "lodash"
 
 export interface NavigationTabsProps {
   searchableConnection: NavigationTabs_searchableConnection$data
   term: string
+  /** TODO: Leave comment explaining why we accept an explicit count for this instead of using the count from the aggregation */
```
Comment (dzucconi) [inline src/Apps/Search/Components/NavigationTabs.tsx:20]: 🤷 Mystery

**Review by nickskalkin — APPROVED** [review]
👍

**Review by damassi — COMMENTED** [review]
beautiful

File: src/Apps/Search/Components/NavigationTabs.tsx (hunk 1 above)
Comment (mzikherman) [reply src/Apps/Search/Components/NavigationTabs.tsx:19]: The artwork count/query is a different query than the `searchConnection` that powers the other tabs, so it likely is queried for and passed in specifically.

https://github.com/artsy/force/blob/25e5863775d92154ce9ddc484875390801055b56/src/Apps/Search/SearchApp.tsx#L153 and https://github.com/artsy/force/blob/25e5863775d92154ce9ddc484875390801055b56/src/Apps/Search/Routes/SearchResultsArtworks.tsx#L65 (as compared to the other entities)

Comment (dzucconi) [reply src/Apps/Search/Components/NavigationTabs.tsx:19]: Why not insert then in mp?

**Pull request merged**